loadtest/control/simulcontroller/controller.go

```diff
@@ -58,8 +59,10 @@ func (c *SimulController) Run() {
 		return
 	}
 
+	semCount := runtime.GOMAXPROCS(0)
+	semaphore := make(chan struct{}, semCount)
 	// Start listening for websocket events.
-	go c.wsEventHandler()
+	go c.wsEventHandler(semaphore)
```

Comment:
I would suggest to keep the semaphore and its logic internal to the handler and here pass a waitgroup.
At this level I think we should just wait for the goroutine of `wsEventHandler` to be finished before shutting down.